app/Services/MarketDataProviders/CoinGecko.php

```diff
@@ -104,6 +106,45 @@ public function priceAndPriceChange(string $baseCurrency, Collection $targetCurr
             ->mapWithKeys(fn (string $currency) => [strtoupper($currency) => MarketData::fromCoinGeckoApiResponse($currency, $data)]);
     }
 
+    /**
+     * @return array{
+     *   price: float,
+     *   volume: float,
+     * }|null
+     */
+    public function exchangeDetails(Exchange $exchange): array | null
+    {
+        $data = null;
+
+        try {
+            $data = Http::get('https://api.coingecko.com/api/v3/exchanges/'.$exchange->coingecko_id.'/tickers', [
+                'coin_ids' => 'ark',
+            ])->json('tickers');
+        } catch (\Throwable) {
+            //
+        }
+
+        if ($this->isEmptyResponse($data) || $this->isThrottledResponse($data)) {
+            throw new ApiNotAvailableException();
+        }
+
+        /** @var array<mixed> $data */
+        $possibleTargets = collect(['USDT', 'USDC', 'BUSD']);
+
+        $tickerData =  collect($data)
+            ->filter(fn (array $ticker) => $possibleTargets->contains($ticker['target']))
+            ->first();
```

Comment:
Can also do this I believe. I'm not sure of the internals but it'll probably stop once it has a match instead of looping over all records

```suggestion
        $tickerData =  collect($data)->first(fn (array $ticker) => $possibleTargets->contains($ticker['target']));
```